### PR TITLE
Add nat_gateways to ec2_repository

### PR DESCRIPTION
**Pull request by lotoussa**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | no
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #165 
| ❓ Documentation  | no

## Description

Add nat_gateway_supplier to ec2_repository to avoid possible hidden dependency.

### CODE REVIEW

**Review by wbeuil — APPROVED** [review]
Nice work, I'm ok with that.

File: pkg/remote/aws/nat_gateway_supplier_test.go [hunk 1]
```diff
@@ -28,50 +28,34 @@ func TestNatGatewaySupplier_Resources(t *testing.T) {
...
-									NatGatewayId: aws.String("nat-0a5408508b19ef490"), // nat1
-								},
-							},
-						}, true)
-						return true
-					})).Return(nil)
+			mocks: func(repo *repository.MockEC2Repository) {
+				repo.On("ListAllNatGateways").Once().Return([]*ec2.NatGateway{
+					{
+						NatGatewayId: aws.String("nat-0bb462a8a4b4bf460"),
```
Comment (eliecharra) [inline pkg/remote/aws/nat_gateway_supplier_test.go:48]: You can reuse the old id and golden file there

**Review by eliecharra — CHANGES_REQUESTED** [review]
Great work thanks 👍🏻

File: pkg/remote/aws/nat_gateway_supplier_test.go (hunk 1 above)
Comment (lotoussa) [reply pkg/remote/aws/nat_gateway_supplier_test.go]: Right.
Change reverted, you can check again.

**Review by wbeuil — APPROVED** [review]

**Pull request merged**